insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/LocalInsightPlugin.java

```diff
@@ -906,21 +900,20 @@ private PumpEnactResult cancelExtendedBolusOnly() {
                     connectionService.requestMessage(cancelBolusMessage).await();
                     confirmAlert(AlertType.WARNING_38);
                     alertService.ignore(null);
-                    InsightBolusID insightBolusID = databaseHelper.getInsightBolusID(connectionService.getPumpSystemIdentification().getSerialNumber(),
-                            activeBolus.getBolusID(), System.currentTimeMillis());
+                    InsightBolusID insightBolusID = insightDatabaseDao.getInsightBolusID(serialNumber(), activeBolus.getBolusID(), dateUtil.now());
                     if (insightBolusID != null) {
-                        ExtendedBolus extendedBolus = databaseHelper.getExtendedBolusByPumpId(insightBolusID.id);
+                        /*  TODO() I don't know if we can remove bolck below (there is a readHistory after that will update AAPS database)
```

Comment:
Can we remove this block ? (I don't update database here because there is a readHistory later, NSUpload is no more necessary, and there is no delete or equivalent in pumpSync interface...)

Reply:
Yes. As far as I remember, this was necessary to properly sync canceled extended boluses if the initial entry was still in the queue.